codalab/lib/bundle_cli.py

```diff
@@ -1173,26 +1188,129 @@ def derive_bundle(self, bundle_type, command, targets, metadata):
 
     @Commands.command(
         'run',
-        help='Create a bundle by running a program bundle on an input bundle.',
+        help='Create a bundle by running a program bundle on an input bundle. If local mode is specified, simulate a run bundle locally, producing bundle contents in the local environment',
         arguments=(
             Commands.Argument('target_spec', help=ALIASED_TARGET_SPEC_FORMAT, nargs='*', completer=TargetsCompleter),
             Commands.Argument('command', metavar='[---] command', help='Arbitrary Linux command to execute.', completer=NullCompleter),
             Commands.Argument('-w', '--worksheet-spec', help='Operate on this worksheet (%s).' % WORKSHEET_SPEC_FORMAT, completer=WorksheetsCompleter),
+            Commands.Argument('--local', action='store_true', help='Simulate a run bundle locally.'),
         ) + Commands.metadata_arguments([RunBundle]) + EDIT_ARGUMENTS + WAIT_ARGUMENTS,
     )
     def do_run_command(self, args):
         client, worksheet_uuid = self.parse_client_worksheet_uuid(args.worksheet_spec)
         args.target_spec, args.command = cli_util.desugar_command(args.target_spec, args.command)
-        targets = self.parse_key_targets(client, worksheet_uuid, args.target_spec)
         metadata = self.get_missing_metadata(RunBundle, args)
-        new_bundle = client.create(
-            'bundles',
-            self.derive_bundle(RunBundle.BUNDLE_TYPE, args.command, targets, metadata),
-            params={'worksheet': worksheet_uuid},
+
+        if args.local:
+            docker_image = metadata.get('request_docker_image', None)
+            if docker_image is None:
```

Comment:
This might need to be a general falsey check, not a `None check: the above call `metadata.get()` will return an empty string if a Docker image is not provided because of how `get_missing_metadata()` works.